### PR TITLE
feat: formalisation of publication filters

### PR DESCRIPTION
**Pull request by reecejohnson**
- [x] Updated Docs

### CODE REVIEW

File: packages/react/src/publication/filters.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+export type PublicationFilters = {
+  restrictPublicationMainFocusTo?: PublicationMainFocus[];
+  restrictPublicationTagsTo?: {
+    all?: string[];
+    oneOf?: string[];
+  };
+  showPublicationsWithContentWarnings?: {
+    oneOf?: PublicationContentWarning[];
+  };
```
Comment (cesarenaldi) [inline packages/react/src/publication/filters.ts:11]: what is the default behaviour when this is not provided?

Comment (paulxuca) [reply packages/react/src/publication/filters.ts:16]: It should not be included in the API request.

**Comment by reecejohnson** [comment]
we also need to add the publication metadata filters to `useComments` 🙏

File: packages/react/src/publication/__tests__/filters.spec.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        },
+      };
+
+      expect(
+        createPublicationMetadataFilters({
+          restrictPublicationTagsTo: {
+            oneOf: ['sports'],
+          },
+        }),
+      ).toEqual(expectedFilters);
```
Comment (reecejohnson) [inline packages/react/src/publication/__tests__/filters.spec.ts:20]: nice 🫡

File: packages/react/src/publication/filters.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import {
+  PublicationContentWarning,
+  PublicationMainFocus,
+  PublicationMetadataFilters,
+} from '@lens-protocol/api-bindings';
+
+export type StandardPublicationMetadataFilter = {
```
[the remark this answers is not in the document]
Comment (paulxuca) [reply packages/react/src/publication/filters.ts:7]: Not sure what to name it without clashing with the existing `PublicationMetadataFilters` from the lens API; Will rename to match the existing type from the API.

**Comment by reecejohnson** [comment]
Look good @paulxuca 👍 

I can't approve though as I opened the original PR

@cesarenaldi can you help here?

**Review by krzysu — APPROVED** [review]
is this documented?

File: packages/react/src/publication/filters.ts (hunk 1 above)
Comment (cesarenaldi) [reply packages/react/src/publication/filters.ts:16]: nit: I believe as of now one can provide and empty object. Is this intentional?
```ts
showPublicationsWithContentWarnings: {}
```

**Review by cesarenaldi — APPROVED** [review]
Small comment.

File: packages/api-bindings/src/graphql/comments.graphql [hunk 4]
```diff
@@ -12,9 +12,16 @@ query Comments(
   $limit: LimitScalar!
   $cursor: Cursor
   $sources: [Sources!]
+  $metadata: PublicationMetadataFilters
```
Comment (desfero) [inline packages/api-bindings/src/graphql/comments.graphql:15]: should we update our field policies to properly set key args? Right now it's will not keep different `metadata` queries as separate items in the apollo cache.

**Review by desfero — CHANGES_REQUESTED** [review]

**Review by desfero — APPROVED** [review]

**Pull request merged**